### PR TITLE
Add vcd_tm_region_storage_policy + vcd_tm_content_library resource and data source

### PR DESCRIPTION
**Pull request by adambarreiro**
Adds data source `vcd_tm_region_storage_policy` Region Storage Policies and `vcd_tm_content_library` data source and resource for Content Libraries.

Some changes are copied from https://github.com/vmware/terraform-provider-vcd/pull/1338 (not pulled), so it should not give conflicts after merged.

**Disclaimer**: There are a lot of unresolved topics marked with `TODO: TM:`, so this implementation will need a revisit.

### CODE REVIEW

File: .github/workflows/check-docs.yml [hunk 1]
```diff
@@ -21,6 +21,9 @@ jobs:
       with:
         fetch-depth: 0  # Required to have tag information available
 
+    - name: Install Terraform binary
+      uses: hashicorp/setup-terraform@v3
+
```
Comment (Didainius) [inline .github/workflows/check-docs.yml:26]: There are a few docs that need fixing

File: go.mod [hunk 2]
```diff
@@ -69,3 +69,5 @@ require (
 	google.golang.org/grpc v1.63.2 // indirect
 	google.golang.org/protobuf v1.34.0 // indirect
 )
+
+replace github.com/vmware/go-vcloud-director/v2 => github.com/adambarreiro/go-vcloud-director/v2 v2.17.0-alpha.1.0.20241011135537-f92ca39ded9c
```
Comment (Didainius) [inline go.mod:73]: Should this be removed now as your PR is merged for govcd

File: vcd/resource_vcd_tm_content_library.go [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+		UpdateContext: resourceVcdTmContentLibraryUpdate,
+		DeleteContext: resourceVcdTmContentLibraryDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdTmContentLibraryImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true, // TODO: Update not supported
```
Comment (Didainius) [inline vcd/resource_vcd_tm_content_library.go:26]: There are many more missed updates, but only mentioning here.

```suggestion
				ForceNew:    true, // TODO: TM: Update not supported
```

File: go.mod (hunk 2 above)
Comment (adambarreiro) [reply go.mod:73]: Indeed, saving this comment as unresolved as reminder.

File: vcd/resource_vcd_tm_content_library.go (hunk 3 above)
Comment (adambarreiro) [reply vcd/resource_vcd_tm_content_library.go]: Fixed, thanks!

**Review by Didainius — APPROVED** [review]
🎉

File: vcd/resource_vcd_tm_content_library_test.go [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+				Config: configText1,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", t.Name()),
+					resource.TestCheckResourceAttr(resourceName, "description", t.Name()),
+					resource.TestCheckResourceAttr(resourceName, "storage_policy_ids.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "auto_attach", "true"), // TODO: TM: Test with false
+					resource.TestCheckResourceAttrSet(resourceName, "creation_date"),
+					resource.TestCheckResourceAttr(resourceName, "is_shared", "true"),        // TODO: TM: Test with false
+					resource.TestCheckResourceAttr(resourceName, "is_subscribed", "false"),   // TODO: TM: Test with true
+					resource.TestCheckResourceAttr(resourceName, "library_type", "PROVIDER"), // TODO: TM: Test with tenant catalog
```
Comment (lvirbalas) [inline vcd/resource_vcd_tm_content_library_test.go:51]: Would it be possible to add these TODO tests in this PR still?

Comment (adambarreiro) [reply vcd/resource_vcd_tm_content_library_test.go:51]: I don't think so, for publishing and subscribe we need the option to be able to publish a catalog, which does not exist yet (only in VCD).
For _tenant_ catalogs I'm not 100% sure we can do it now, but I'll investigate on next PRs.

Comment (lvirbalas) [reply vcd/resource_vcd_tm_content_library_test.go:51]: Got it. Thanks!

**Review by lvirbalas — APPROVED** [review]

**Pull request merged**